### PR TITLE
Create `Settings` value object for `$cfg` global

### PR DESCRIPTION
**Pull request by MauricioFauth**
This will improve type checking for `$cfg` values.

Example (note the `int` type cast being removed):
```diff
diff --git a/libraries/classes/Html/Generator.php b/libraries/classes/Html/Generator.php
index 147adc6b42..fe0567a2d9 100644
--- a/libraries/classes/Html/Generator.php
+++ b/libraries/classes/Html/Generator.php
@@ -447,7 +447,8 @@ class Generator
         ) {
             $rows = (int) $_SESSION['tmpval']['max_rows'];
         } else {
-            $rows = (int) $GLOBALS['cfg']['MaxRows'];
+            $settings = new \PhpMyAdmin\Config\Settings($GLOBALS['cfg']);
+            $rows = $settings->MaxRows;
             $_SESSION['tmpval']['max_rows'] = $rows;
         }
```

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
This is, incredible!
💯 🏅

File: libraries/classes/Config/Settings.php [hunk 1]
```diff
@@ -0,0 +1,2450 @@
...
+     * @var int
+     */
+    public $MaxRows = 25;
+
+    /**
+     * default for 'ORDER BY' clause (valid values are 'ASC', 'DESC' or 'SMART' -ie
+     * descending order for fields of type TIME, DATE, DATETIME & TIMESTAMP,
+     * ascending order else-)
+     *
+     * @var string
```
Comment (williamdes) [inline libraries/classes/Config/Settings.php:773]:
```suggestion
     * @var string
     * @phpstan-var 'ASC'|'DESC'|'SMART'
```

File: libraries/classes/Config/Settings/Schema.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+// phpcs:disable Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+
+/**
+ * @psalm-immutable
+ */
+final class Schema
+{
+    /**
+     * pdf/eps/dia/svg
```
Comment (williamdes) [inline libraries/classes/Config/Settings/Schema.php:15]: psalm-var :)

**Review by williamdes — APPROVED** [review]

**Comment by mauriciofauth** [comment]
I created a basic page to check `config.inc.php` for errors using the `Settings` class. For now it is just a prototype.

Screenshot_2021-07-30 localhost 8000 localhost phpMyAdmin 5 2 0-dev

**Comment by ibennetch** [comment]
Great! Seems like this is a good idea and positive change.

**Comment by williamdes** [comment]
👏🏻 This will save us from bugs!
Awesome!
Will you support deprecated/removed options (example: old config to choose mysqli extension)

**Comment by mauriciofauth** [comment]
> 👏🏻 This will save us from bugs!
> Awesome!
> Will you support deprecated/removed options (example: old config to choose mysqli extension)
> I mean to say to people they can or should remove old non used keys 

That is exactly what I was thinking about. It will help users with deprecated/removed/renamed settings. And with different than expected values.

**Comment by mauriciofauth** [comment]
> I created a basic page to check `config.inc.php` for errors using the `Settings` class. For now it is just a prototype.

I will open a new pull request with this page.

**Pull request merged**